docs/latest/design/extending-envoy-gateway.md

```diff
@@ -144,6 +144,62 @@ latency between Envoy and the authentication service and also allows the end use
 
 ![Control Plane Extension Example](../images/extension-auth-example.png)
 
+## Design Decisions
+
+- Since an HTTPRoute resource corresponds to an [Envoy Proxy Route resource][], the name of the Route is set using the name and namespace of the HTTPRoute that generated it so that any extension looking at the
+xDS resources provided by Envoy Gateway can have the same context about which Routes to modify.
+- The responsibility of managing scale and performance of hook calls and modification of xDS resources falls on the extension controller, not Envoy Gateway
```

Comment:
I'm not sure I 100% agree with this with the way it is written just because it has no context in this scenario. But to me its generally speaking: 

- Extension Server takes ownership of producing the correct configuration ( _proper xDS config or proper IR config filled in_) in the hook response
- Extension Server will be responsible for ensuring the performance of the hook processing time


Then after that I think @haq204 comment below, discussing the trade-off of using unary vs streaming is probably warrants some additional input.